SIPS/sip-29.md

```diff
@@ -243,9 +267,126 @@ type FungibleAssetMetadata = {
   // List of asset units.
   units: FungibleAssetUnit[];
 };
+```
+### Asset Market Data
 
-// Represents the metadata of an asset.
-type AssetMetadata = FungibleAssetMetadata
+```typescript
+type FungibleAssetMarketData = {
+  // Represents a fungible asset market data.
+  fungible: true;
+
+  // The market cap of the asset represented as a decimal number in a string.
+  marketCap?: string;
+
+  // The total volume of the asset represented as a decimal number in a string.
+  totalVolume?: string;
+
+  // The circulating supply of the asset represented as a decimal number in a string.
+  circulatingSupply?: string;
+
+  // The all time high of the asset represented as a decimal number in a string.
+  allTimeHigh?: string;
+
+  // The all time low of the asset represented as a decimal number in a string.
+  allTimeLow?: string;
+
+  pricePercentChange?: {
+    // The `all` value is a special interval that represents all available data.  
+    all?: number;  
+
+    // The interval key MUST follow the ISO 8601 duration format.  
+    [interval: string]: number;
+  };
+};
+```
+
+## Appendix II: Non-fungible Asset Metadata & Market Data
+
+The following asset metadata fields for a non-fungible asset are defined.
+
+### Asset Metadata
+```typescript
+type NonFungibleAssetCollection = {
+  // Human-friendly name of the asset collection.
+  name: string;
+
+  // Ticker symbol of the asset collection.
+  symbol: string;
+
+  // The number of tokens in the collection.
+  tokenCount: string;
+
+  // The creator address of the asset.
+  creator?: Caip19AssetType;
+
+  // Base64 data URI or URL representation of the asset icon.
+  imageUrl: string;
+};
+
+type NonFungibleAssetMetadata = {
+  // Human-friendly name of the asset.
+  name?: string;
+
+  // Ticker symbol of the asset.
+  symbol?: string;
+
+  // Base64 data URI or URL representation of the asset image.
+  imageUrl: string;
+
+  // The description of the asset.
+  description?: string;
```

Comment:
I believe we also want to add
```
isSpam?: boolean
acquired_at?: string
```

We use blockaid as a safety net for spam filtering, would be used on controller side to filter out spam nfts;

Reply:
for `isSpam` I would say the snap can handle that filtering, this seems like a very "only us" related field

Agree that `acquired_at` needs to be included !